nucliadb_utils/nucliadb_utils/storages/pg.py

```diff
@@ -0,0 +1,560 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import annotations
+
+import asyncio
+import uuid
+from typing import Any, AsyncIterator, Optional, TypedDict
+
+import asyncpg
+from nucliadb_protos.resources_pb2 import CloudFile
+
+from nucliadb_utils.storages import CHUNK_SIZE
+from nucliadb_utils.storages.storage import Storage, StorageField
+
+# Table design notes
+# - No foreign key constraints ON PURPOSE
+# - No cascade handling ON PURPOSE
```

Comment:
Out of curiosity: What are the implications? This assuming that kb_id conflicts will be handled in upper layers, right?

Reply:
The reason is really only performance.

I can add them in if we don't care and we'd have a little more guarantee on data quality.

I don't mind either way!